ArduCopter/mode_smart_rtl.cpp

```diff
@@ -174,8 +181,12 @@ void ModeSmartRTL::pre_land_position_run()
 // save current position for use by the smart_rtl flight mode
 void ModeSmartRTL::save_position()
 {
-    const bool should_save_position = motors->armed() && (copter.flightmode->mode_number() != Mode::Number::SMART_RTL);
+    // Prevent saving positions when the drone is disarmed
+    if (!motors->armed()) {
```

Comment:
We've already got a "should_save_position" below so I don't think we need this extra armed check.

I suspect the issue is not that a point is being saved but that home is being saved while the vehicle is disarmed.  This means that if the user moves the vehicle after arming SmartRTL may move to the original position.  Could you confirm this is the issue you're seeing?

Reply:
Yes this is the issue i was seeing so that it should not go to smartrtl mode before arming the copter that's why i added that thanks for reviewing!

Reply:
Hi @harsh839,

OK thanks for that.  I think a better solution then would be to modify the AP_SmartRTL::update() function so that it takes a "save_home" argument that is set to true if the vehicle is disarmed.

Reply:
Actually after looking at the code a bit more, I see that when the vehicle arms it calls copter.g2.smart_rtl.set_home(copter.position_ok()) so there shouldn't be any need to do what I've mentioned above.

I'm sorry @harsh839, I don't see the problem.  Are you sure that you've reproduced a problem and shown that this PR fixes it?  Could you please share some logs or screen shots showing the before and after?

Reply:
ohk i will be posting here just check my another PR which fixes #17419